libr/io/undo.c

```diff
@@ -170,21 +163,27 @@ R_API RList *r_io_sundo_list(RIO *io, int mode) {
 			if (list) {
 				RIOUndos  *u = R_NEW0 (RIOUndos);
 				if (u) {
-					memcpy (u, undo, sizeof (RIOUndos));
+					if (!(j == undos && redos == 0)) {
+						// Current position gets pushed before seek, so there
+						// is no valid offset when we are at the end of list.
+						memcpy (u, undo, sizeof (RIOUndos));
```

Comment:
wlll be good to abstract thihs inito a function named r_io_undos_clone(), but with the comment i guess its fine too